### ISSUE TITLE
MA0004 conflicts with xUnit1030

### ISSUE THREAD

**Issue by samtrion** [meziantou/Meziantou.Analyzer#678]
### Version of the Meziantou.Analyzer NuGet package

2.0.133

### Rule Identifier

MA0004

### Target Framework

NET8.0

### C# Language version

C# 12

### Description

In xUnit test projects, there are contradictory statements between MA0004 and [xUnit1030](https://xunit.net/xunit.analyzers/rules/xUnit1030).

### Reproduction Steps

Minimal code:

```c#
public async Task Test_Async()
{
    await Task.Delay(1); // MA0004 would like to add `.ConfigureAwait(false)`, xUnit1030 would like to remove it or replace it with `.ConfigureAwait(true)`.
    Assert.True(true);
}
```

**Comment by samtrion** [comment]
nvm

**Comment by meziantou** [comment]
MA0004 should not reported in methods decorated by [Fact] or [Theory]. If it's reported, please open a new issue with a repro.

**Comment by samtrion** [comment]
I have now been able to reproduce the error. [dailydevops/healthchecks](https://github.com/dailydevops/healthchecks/blob/error/MA0004/src/HealthChecks/tests/NetEvolve.HealthChecks.Tests.Unit/ApplicationSelfCheckTests.cs#L15-L26)

![image](https://github.com/meziantou/Meziantou.Analyzer/assets/3283596/650d83bf-e692-4f14-9b87-10d2d25ab521)

**Comment by meziantou** [comment]
When you create a local function, it's actually another method. Your example is not different from 

````c#
public class Test
{
    // xunit1030 and MA0004 are not reported
    [Fact]
    public async Task Test01() => await Act();

    // xunit1030 is not reported
    async Task Act() => await Sample.Method().ConfigureAwait(false);
}

class Sample
{
    public static async Task Method() => throw null;
}
````

So, I think it's a xUnit1030 bug. You should open an issue on the xunit repository. If xunit tells it's the expected behavior, you can link the issue and reopen this issue.
